File: ios/Classes/lib/notifications/managers/CreatedManager.swift
//
//  CreatedManager.swift
//  awesome_notifications
//
//  Created by Rafael Setragni on 15/09/20.
//

import Foundation

public class CreatedManager {
    
    static let shared:SharedManager = SharedManager(tag: Definitions.SHARED_CREATED)
    
    public static func removeCreated(id:Int) -> Bool {
        return shared.remove(referenceKey: String(id));
    }

    public static func listCreated() -> [NotificationReceived] {
        var returnedList:[NotificationReceived] = []
        let dataList = shared.getAllObjects()
        
        for data in dataList {
            let received:NotificationReceived = NotificationReceived(nil).fromMap(arguments: data) as! NotificationReceived
            returnedList.append(received)
        }
        
        return returnedList
    }

    public static func saveCreated(received:NotificationReceived) {
        shared.set(received.toMap(), referenceKey: String(received.id!))
    }

    public static func getCreatedByKey(id:Int) -> NotificationReceived? {
        guard let data:[String:Any?] = shared.get(referenceKey: String(id)) else {
          return nil
        }
        return NotificationReceived(nil).fromMap(arguments: data) as? NotificationReceived
    }

    public static func cancelAllCreated() {
        shared.removeAll()
    }

    public static func cancelCreated(id:Int) {
        _ = shared.remove(referenceKey: String(id))
    }
    
}

File: android/src/main/java/me/carda/awesome_notifications/notifications/managers/CreatedManager.java
package me.carda.awesome_notifications.notifications.managers;

import android.content.Context;

import com.google.common.reflect.TypeToken;

import java.lang.reflect.Type;
import java.util.List;

import me.carda.awesome_notifications.AwesomeNotificationsPlugin;
import me.carda.awesome_notifications.Definitions;
import me.carda.awesome_notifications.notifications.models.returnedData.NotificationReceived;

public class CreatedManager {

    private static final SharedManager<NotificationReceived> shared
            = new SharedManager<>(
                    "CreatedManager",
                    NotificationReceived.class,
                    "NotificationReceived");

    public static Boolean removeCreated(Context context, Integer id) {
        return shared.remove(context, Definitions.SHARED_CREATED, id.toString());
    }

    public static List<NotificationReceived> listCreated(Context context) {
        return shared.getAllObjects(context, Definitions.SHARED_CREATED);
    }

    public static void saveCreated(Context context, NotificationReceived received) {
        shared.set(context, Definitions.SHARED_CREATED, received.id.toString(), received);
    }

    public static NotificationReceived getCreatedByKey(Context context, Integer id){
        return shared.get(context, Definitions.SHARED_CREATED, id.toString());
    }

    public static void cancelAllCreated(Context context) {
        List<NotificationReceived> receivedList = shared.getAllObjects(context, Definitions.SHARED_CREATED);
        if (receivedList != null){
            for (NotificationReceived received : receivedList) {
                cancelCreated(context, received.id);
            }
        }
    }

    public static void cancelCreated(Context context, Integer id) {
        NotificationReceived received = getCreatedByKey(context, id);
        if(received !=null)
            removeCreated(context, received.id);
    }

    public static void commitChanges(Context context){
        shared.commit(context);
    }
}
